core/commands/VersionDeleteCommand.php

```diff
@@ -0,0 +1,118 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+require_api( 'constant_inc.php' );
+require_api( 'config_api.php' );
+require_api( 'helper_api.php' );
+require_api( 'version_api.php' );
+
+use Mantis\Exceptions\ClientException;
+
+/**
+ * A command that adds a project version.
+ */
+class VersionDeleteCommand extends Command {
+	/**
+	 * The project id
+	 * @var integer
+	 */
+	private $project_id;
+
+	/**
+	 * The version id
+	 * @var integer
+	 */
+	private $version_id;
+
+	/**
+	 * $p_data['query'] is expected to contain:
+	 * - project_id (integer)
+	 * - version_id (integer)
+	 *
+	 * $p_data['payload'] is expected to contain:
+	 * - name (string)
+	 * - description (string)
+	 * - released (bool)
+	 * - obsolete (bool)
+	 * - timestamp (timestamp) - e.g. 2018-05-21T15:00:00-08:00
+	 *
+	 * @param array $p_data The command data.
+	 */
+	function __construct( array $p_data ) {
+		parent::__construct( $p_data );
+	}
+
+	/**
+	 * Validate the data.
+	 */
+	function validate() {
+		$this->project_id = helper_parse_id( $this->query( 'project_id' ), 'project_id' );
+
+		if( !project_exists( $this->project_id ) ) {
+			throw new ClientException(
+				"Project '$this->project_id' not found",
+				ERROR_PROJECT_NOT_FOUND,
+				array( $this->project_id ) );
+		}
+
+		$t_access_level = config_get(
+			'manage_project_threshold',
+			/* default */ null,
+			/* user */ null,
+			$this->project_id );
+		if( !access_has_project_level( $t_access_level, $this->project_id ) ) {
+			throw new ClientException( 'Access denied to delete version', ERROR_ACCESS_DENIED );
+		}
+
+		$this->version_id = helper_parse_id( $this->query( 'version_id' ), 'version_id' );
+		if( !version_exists( $this->version_id ) ) {
+			return;
```

Comment:
@vboctor is it intentional that specifiying a non-existing version is not triggering an ERROR_VERSION_NOT_FOUND ClientException ? This results in the API returning a 204 (version deleted)

Reply:
Proposed fix in PR #1905

Reply:
I'll comment on the proposed fix PR.